geps/gep-1911.md

```diff
@@ -0,0 +1,207 @@
+
+Currently Kubernetes `Service` API does not allow different `appProtocol` values for the same port. At this time there seems to be interest in changing `appProtocol` to be a list in order to faciliate this use-case.
+
+Open Question: Can SCTP be multiplexed with TCP and UDP on the same port?
+
+### Unsupported/Incompatible Protocols
+
+If a Route attached to a Gateway is not able to send traffic to the backend using the specified protocol then the backend is considered invalid. Implementations MUST set `ResolvedRefs` condition to `False` with the Reason `UnsupportedProtocol`.
+
+It is RECOMMENDED that implementaions follow the supported combinations below:
+
+ServicePort Protocol | Route Type | Supported
+-|-|-
+`UDP`  | TCPRoute  | No
+`SCTP` | TCPRoute  | No
+`UDP`  | TLSRoute  | Conditional [1]
+`SCTP` | TLSRoute  | No
+`TCP`  | UDPRoute  | No
+`SCTP` | UDPRoute  | No
+`SCTP` | GRPCRoute | No
+`SCTP` | HTTPRoute | No
+
+1. Open Question: TLS over UDP seems to be a thing via QUIC/HTTP3 [ref](https://www.smashingmagazine.com/2021/08/http3-core-concepts-part1/).
+Likewise there's also [DTLS](https://en.wikipedia.org/wiki/Datagram_Transport_Layer_Security). But it's unclear if Gateway's TLSRoute
+applies to an underlying UDP protocol.
+
+ServicePort Protocol | ServicePort AppProtocol | Supported
+-|-|- 
+`UDP`  |`kubernetes.io/h2c` | No
+`SCTP` |`kubernetes.io/h2c` | No
+`UDP`  |`kubernetes.io/ws`  | No [1]
+`SCTP` |`kubernetes.io/ws`  | No
+`UDP`  |`kubernetes.io/wss` | No [1]
+`SCTP` |`kubernetes.io/wss` | No
+
+1. Open Question: Should we upstream new constants for websocket over [HTTP/2](https://www.rfc-editor.org/rfc/rfc8441.html)
+    & [HTTP/3](https://www.rfc-editor.org/rfc/rfc9220.html) ?
+
+ServicePort Protocol | ServicePort AppProtocol | Route Type | Supported
+-|-|-|-
+`TCP`  |`kubernetes.io/h2c` | `HTTPRoute` | Yes
+`UDP`  |`kubernetes.io/h2c` | `HTTPRoute` | No
+`TCP`  |`kubernetes.io/h2c` | `GRPCRoute` | Yes [2]
+`UDP`  |`kubernetes.io/h2c` | `GRPCRoute` | No
+`UDP`  |`kubernetes.io/h2c` | `UDPRoute`  | No
+`TCP`  |`kubernetes.io/h2c` | `TCPRoute`  | TBD [4]
+`TCP`  |`kubernetes.io/h2c` | `TLSRoute`  | Conditional [1]
+`TCP`  |`kubernetes.io/ws`  | `HTTPRoute` | Yes
+`UDP`  |`kubernetes.io/ws`  | `HTTPRoute` | TBD [3]
+`TCP`  |`kubernetes.io/ws`  | `GRPCRoute` | No
+`UDP`  |`kubernetes.io/ws`  | `GRPCRoute` | No
+`TCP`  |`kubernetes.io/ws`  | `TCPRoute`  | TBD [4]
+`TCP`  |`kubernetes.io/ws`  | `TLSRoute`  | Conditional [1]
+`TCP`  |`kubernetes.io/wss` | `HTTPRoute` | Conditional [1]
+`UDP`  |`kubernetes.io/wss` | `HTTPRoute` | TBD [3]
+`TCP`  |`kubernetes.io/wss` | `GRPCRoute` | No
+`UDP`  |`kubernetes.io/wss` | `GRPCRoute` | No
+`TCP`  |`kubernetes.io/wss` | `TCPRoute`  | TBD [4]
+`TCP`  |`kubernetes.io/wss` | `TLSRoute`  | Conditional [1]
+
+1. Only if there is a corresponding `TLSConnectionPolicy` - see [GEP-1897](TODO)
+2. GRPC works over h2c
```

Comment:
Is it worth calling out that doing unencrypted gRPC over an external connection like a Gateway is pretty weird though?

Reply:
> Is it worth calling out that doing unencrypted gRPC over an external connection like a Gateway is pretty weird though?

I just wanted to highlight GRPC can happen over clear text. I updated the clause to be more specific that the clear text connection is from the gateway to the backend.